conan/tools/apple/apple.py

```diff
@@ -57,39 +61,52 @@ def get_apple_sdk_fullname(conanfile):
         raise ConanException("Please, specify a suitable value for os.sdk.")
 
 
-def apple_min_version_flag(os_version, os_sdk, subsystem):
+def apple_min_version_flag(conanfile):
     """compiler flag name which controls deployment target"""
-    if not os_version or not os_sdk:
-        return ''
-
-    # FIXME: This guess seems wrong, nothing has to be guessed, but explicit
-    flag = ''
-    if 'macosx' in os_sdk:
-        flag = '-mmacosx-version-min'
-    elif 'iphoneos' in os_sdk:
-        flag = '-mios-version-min'
-    elif 'iphonesimulator' in os_sdk:
-        flag = '-mios-simulator-version-min'
-    elif 'watchos' in os_sdk:
-        flag = '-mwatchos-version-min'
-    elif 'watchsimulator' in os_sdk:
-        flag = '-mwatchos-simulator-version-min'
-    elif 'appletvos' in os_sdk:
-        flag = '-mtvos-version-min'
-    elif 'appletvsimulator' in os_sdk:
-        flag = '-mtvos-simulator-version-min'
-    elif 'xros' in os_sdk:
+    os_ = conanfile.settings.get_safe('os')
+    os_sdk = conanfile.settings.get_safe('os.sdk')
+    os_sdk = os_sdk or ("macosx" if os_ == "Macos" else None)
+    os_version = conanfile.settings.get_safe("os.version")
+    subsystem = conanfile.settings.get_safe("os.subsystem")
+    if os_sdk is None or not os_version:
+        raise ConanException("Please, specify a suitable value for os.sdk")
+    flag = {
+        "macosx": "-mmacosx-version-min",
+        "iphoneos": "-mios-version-min",
+        "iphonesimulator": "-mios-simulator-version-min",
+        "watchos": "-mwatchos-version-min",
+        "watchsimulator": "-mwatchos-simulator-version-min",
+        "appletvos": "-mtvos-version-min",
+        "appletvsimulator": "-mtvos-simulator-version-min",
         # need early return as string did not fit into the "normal" schema
-        return f'-target arm64-apple-xros{os_version}'
-    elif 'xrsimulator' in os_sdk:
+        "xros": f"-target arm64-apple-xros{os_version}",
         # need early return as string did not fit into the "normal" schema
-        return f'-target arm64-apple-xros{os_version}-simulator'
+        "xrsimulator": f"-target arm64-apple-xros{os_version}-simulator",
+    }.get(os_sdk,
+          # especial case, despite Catalyst is macOS, it requires an iOS version argument
+          '-mios-version-min' if subsystem == 'catalyst' else '')
```

Comment:
This is changing behavior, previously the ``catalyst`` was overriding the ``macos``, but no longer the case with this change